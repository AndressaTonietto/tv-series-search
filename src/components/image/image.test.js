import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import Image from "./image";
import renderer from "react-test-renderer";

const testAlt = "test alt text";
const testSrc = "test-src.jpg";

describe("Image", () => {
  it("renders correctly with src", () => {
    const tree = renderer
      .create(<Image alt={testAlt} src={testSrc} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly without src", () => {
    const tree = renderer
      .create(<Image alt={testAlt} src={undefined} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the image and the alt text correctly when src is provided", () => {
    render(<Image alt={testAlt} src={testSrc} />);

    const imgElement = screen.getByRole("img");
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src", testSrc);
    expect(imgElement).toHaveAttribute("alt", testAlt);
  });

  it("renders 'No image available' when src is not provided", () => {
    render(<Image alt={testAlt} src={undefined} />);

    expect(screen.getByText("No image available")).toBeInTheDocument();
  });
});
