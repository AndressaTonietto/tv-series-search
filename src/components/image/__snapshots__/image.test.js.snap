// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly with src 1`] = `
<img
  alt="test alt text"
  className="max-w-52"
  src="test-src.jpg"
/>
`;

exports[`Image renders correctly without src 1`] = `
<div
  className="aspect-[2/3] justify-center max-w-52 flex items-center px-2 badge-neutral"
>
  No image available
</div>
`;
